### PR TITLE
Diary work

### PR DESCRIPTION
**Pull request by seth-at-at**
Updated seed file, finished endpoints for diary.

### CODE REVIEW

File: db/seeds/dev/users.js [hunk 1]
```diff
@@ -4,15 +4,27 @@ exports.seed = function(knex, Promise) {
     return Promise.all([
```
Comment (tmikeschu) [inline db/seeds/dev/users.js:4]: The file name is still users...

File: lib/models/diary.js [hunk 2]
```diff
@@ -1,35 +1,61 @@
...
-  first: findFirst,
+  destroyAll: emptyDiarysTable,
   find: find,
-  findName: findName,
-  destroyAll: emptyDiarysTable
-}
+  delete: deleteSelected,
+  updateName: updateName,
+  // findFoodsFor: findFoodsFor,
+  all: all,
```
Comment (tmikeschu) [inline lib/models/diary.js:59]: when the key and value have the same name, you can just say it once like: `all,`

Comment (tmikeschu) [reply lib/models/diary.js:59]: I'd also like to see all of these having the same names.

File: server.js [hunk 3]
```diff
@@ -32,6 +32,33 @@ app.post('/api/v1/diary', (request, response) => {
...
+    })
+});
+
+app.put('/api/v1/diary/:id', (request, response) => {
+  let diaryId = request.params.id
+  let diaryName = request.body.name
+
+  Diary.updateName(diaryName, diaryId)
+    .then( (data) => {
+    if (data.rowCount === 0) { return response.sendStatus(404) }
```
Comment (tmikeschu) [inline server.js]: I'm not sure 404 is the status you want to send. They "found" an endpoint, but their request was somehow bad. For "bad requests", opt for 400.

File: test/diary-test.js [hunk 4]
```diff
@@ -14,15 +14,58 @@ describe('Server', function() {
...
+  describe('methods', function(){
+    beforeEach( function(done){
+      Diary.create('test')
+        .then(() => done());
+    });
+
+    afterEach( function(done){
+      Diary.destroyAll().then(() => done());
+    });
+    // create: createDiary,
```
Comment (tmikeschu) [inline test/diary-test.js:34]: Please get rid of these.

File: test/diary-test.js [hunk 5]
```diff
@@ -14,15 +14,58 @@ describe('Server', function() {
...
+    // destroyAll: emptyDiarysTable,
+    // find: find,
+    // findFoodsFor: findFoodsFor,
+    // all: all,
+    // findByName: findByName
+    it('Creates diary / #find', function(done){
```
Comment (tmikeschu) [inline test/diary-test.js:40]: Is a diary created?

File: test/diary-test.js [hunk 6]
```diff
@@ -45,13 +88,12 @@ describe('Server', function() {
 
     it('should have the id and message from the resource', function(done){
       let ourRequest = this.request
```
Comment (tmikeschu) [inline test/diary-test.js:90]: I don't see an actual assertion in this test...

File: test/diary-test.js [hunk 7]
```diff
@@ -81,19 +124,75 @@ describe('Server', function() {
...
+      let lunch = { name: 'lunch' }
+      let options = {
+        body: lunch,
+        json: true,
+        url: '/diary/1'
+      }
+      this.request.delete(options, (error, response, body) => {
+        if (error) { done(error) }
+
+        assert.equal(body.name, undefined)
```
Comment (tmikeschu) [inline test/diary-test.js:166]: Does this verify that the item was deleted from the database?

**Review by tmikeschu — COMMENTED** [review]
@bschwartz10  @seth-at-at see comments

File: lib/models/diary.js (hunk 2 above)
Comment (bschwartz10) [reply lib/models/diary.js:59]: I'm not exactly sure what you mean by "you can just say it once like: `all` ".

Also is it convention to have the key/value be the same?

Comment (tmikeschu) [reply lib/models/diary.js:59]: Sorry about the ambiguity. If the key and value are the same name (the value being a variable), you can convert:

```js
const obj =   {
  thing: thing,
  otherThing: otherThing
}
```

to:

```js
const obj = {
  thing,
  otherThing
}
```

**Pull request merged**